README.md

```diff
@@ -1,138 +1,134 @@
-
-- pullconsumer must invoke following interface:
-  - DefaultMQPullConsumer     g_consumer("please_rename_unique_group_name");
-  - g_consumer.start();
-  - g_consumer.fetchSubscribeMessageQueues(..., ...);
-  - g_consumer.pull(...)
-  - g_consumer.shutdown();
-
-- pushconsumer must invoke following interface:
-  - DefaultMQPushConsumer g_consumer("please_rename_unique_group_name_1");
-  - g_consumer.subscribe("test_topic", "*");
-  - g_consumer.registerMessageListener(listener);
-  - g_consumer.start();
-  - g_consumer.shutdown();
-  
+# RocketMQ-Client-CPP #
+
+RocketMQ-Client-CPP is the C/C++ client of Apache RocketMQ which is a distributed messaging and streaming platform with low latency, high performance and reliability, trillion-level capacity and flexible scalability.
+
+## Features ##
+
+- produce message, include sync and asynce produce message, timed and delay message. 
+
+- consume message, include concurrency and orderly consume message, broadcast and cluster consume model.
+
+- rebalance, include both produce message and consume message with rebalance.
+
+- C and C++ API, include both C style and C++ style apis.
+
+- across platform, all features are supported on both windows and linux system.
+
+- low latency, publish latency < 2ms, subscribe latency < 10ms
+
+- high tps, rocketmq topic with 16 message queues, stand-alone cpp client can reach publish TPS > 3w and subsricbe TPS > 15w.
+
+- reliability, based on nameServer snapshot and network disaster recovery strategy, no real-time impact on publish and subscribe when anyone of broker or nameSrv was broken.
+
+## Dependency ##
+- [libevent 2.0.22](https://github.com/libevent/libevent/archive/release-2.0.22-stable.zip "libevent 2.0.22")
+
+- [jsoncpp 0.10.6](https://github.com/open-source-parsers/jsoncpp/archive/0.10.6.zip  "jsoncpp 0.10.6")
+
+- [boost 1.56.0](http://sourceforge.net/projects/boost/files/boost/1.56.0/boost_1_56_0.tar.gz "boost 1.56.0")
+
+## Documentation ##
+doc/rocketmq-cpp_manaual_zh.docx
+
+## Build and Install ##
+
+### Linux platform ###
+
+**note**: *make sure the following compile tools or libraries with them minimum version number have been installed before run the build script build.sh*
+
+- compile tools:
+	- gcc-c++ 4.8.2: jsoncpp,boost rocket-client require it, need support C++11
+	- cmake 2.8.0: jsoncpp,rocketmq-client require it
+	- automake 1.11.1: libevent require it
+	- libtool 2.2.6: libevent require it
+
+- libraries:   
+	- bzip2-devel 1.0.6: boost dependcy it
+
+one key build script will automatic build the dependency libraries include libevent json and boost, then it will build rocketmq-client static and shared library.
+
+if can't get internet to download three library source files by build script, you can copy three library source files (release-2.0.22-stable.zip  0.10.6.zip and boost_1_56_0.tar.gz) to rocketmq-client root dir, then build.sh will auto use these library files to build rocketmq-client.
+
+    sudo sh build.sh

```

Comment:
Is it still compiled with build.sh? Is the compilation logic of onekeybuild.sh merged to build.sh?

Reply:
yes， onekeybuild.sh change name to build.sh